prebuilt_swift_static_framework/prebuilt_swift_static_framework.bzl

```diff
@@ -21,40 +27,73 @@ def _zip_swift_arg(module_name, swift_identifier, input_file):
-    module_name = ctx.attr.framework_name
+    module_name = ctx.attr.module_name
     fat_file = ctx.outputs.fat_file
 
-    input_archives = []
-    input_modules_docs = []
-    zip_args = [_zip_binary_arg(module_name, fat_file)]
+    input_libraries = []
+    input_swift_info = []
+    generated_objc_hdr_file = None
+
+    # The Swift static framework has only one `swift_library` dependency target.
+    platform_to_target_list = ctx.split_attr.archive.items()
+    swift_library_target = platform_to_target_list[0][1]
+
+    # Get the generated Objective-C header file for the `swift_library`.
+    objc_provider = swift_library_target[apple_common.Objc]
+    generated_hdr_file_basename = swift_library_target.label.name + "-Swift.h"
+    for objc_hdr_file in objc_provider.header.to_list():
+        if objc_hdr_file.basename == generated_hdr_file_basename:
+            generated_objc_hdr_file = objc_hdr_file
+            break
+
+    zip_args = [
+        _zip_binary_arg(module_name, fat_file),
+        _zip_generated_objc_hdr_arg(module_name, generated_objc_hdr_file),
+    ]
 
-    for platform, archive in ctx.split_attr.archive.items():
+    # Get the `swiftdoc` and `swiftmodule` files for each iOS platform.
+    for platform, target in platform_to_target_list:
         swiftmodule_identifier = _PLATFORM_TO_SWIFTMODULE[platform]
-        library = archive[CcInfo].linking_context.libraries_to_link[0].pic_static_library
-        swift_info = archive[SwiftInfo]
-        swiftdoc = swift_info.direct_swiftdocs[0]
-        swiftmodule = swift_info.direct_swiftmodules[0]
+        if not swiftmodule_identifier:
+            continue
+
+        # Bazel 0.27 changed `libraries_to_link` from a list to a `depset`.
+        libraries_to_link = target[CcInfo].linking_context.libraries_to_link
+        if hasattr(libraries_to_link, "to_list"):
+            libraries_to_link = libraries_to_link.to_list()
+        library = libraries_to_link[0].pic_static_library
+        input_libraries.append(library)
 
-        input_archives.append(library)
-        input_modules_docs += [swiftdoc, swiftmodule]
+        swift_info_provider = target[SwiftInfo]
+        swiftdoc = swift_info_provider.direct_swiftdocs[0]
+        swiftmodule = swift_info_provider.direct_swiftmodules[0]
+        input_swift_info += [swiftdoc, swiftmodule]
         zip_args += [
             _zip_swift_arg(module_name, swiftmodule_identifier, swiftdoc),
             _zip_swift_arg(module_name, swiftmodule_identifier, swiftmodule),
         ]
 
-    ctx.actions.run(
-        inputs = input_archives,
+    apple_support.run(
+        ctx,
+        inputs = input_libraries,
         outputs = [fat_file],
         mnemonic = "LipoSwiftLibraries",
         progress_message = "Combining libraries for {}".format(module_name),
-        executable = "lipo",
-        arguments = ["-create", "-output", fat_file.path] + [x.path for x in input_archives],
+        executable = "/usr/bin/lipo",
```

Comment:
I wonder if this should be `xcrun`, and then `arguments` be `["lipo", "-create", ...]`.

Reply:
Good call. That should work.